src/core/engines/Cesium/Feature/index.tsx

```diff
@@ -70,56 +73,85 @@ const pickProperty = (k: keyof AppearanceTypes, layer: ComputedLayer) => {
   return layer.layer[k];
 };
 
+const CACHED_COMPONENTS = new LRUCache<string, JSX.Element>({ max: 10000 });
+const FEATURE_DELEGATE_THRESHOLD = 6000;
+
 export default function Feature({
   layer,
   isHidden,
   ...props
 }: FeatureComponentProps): JSX.Element | null {
+  // Set the displayConfig based on the size of the feature collection array
+  if (layer.features?.length > FEATURE_DELEGATE_THRESHOLD) {
+    displayConfig.geojson = ["resource"];
+  } else {
+    displayConfig.geojson = "auto";
+  }
   const data = extractSimpleLayerData(layer);
   const ext = !data?.type || (data.type as string) === "auto" ? guessType(data?.url) : undefined;
   const displayType = data?.type && displayConfig[ext ?? data.type];
   const areAllDisplayTypeNoFeature =
     Array.isArray(displayType) &&
     displayType.every(k => components[k][1].noFeature && !components[k][1].noLayer);
+  const cachedRenderComponent = (
+    k: keyof AppearanceTypes,
+    f?: ComputedFeature,
+  ): JSX.Element | null => {
+    const componentId = `${layer.id}_${f?.id ?? ""}_${k}`;
 
-  const renderedComponents = new Set<string>(); // Initialize a set to store rendered components
-
-  const renderComponent = (k: keyof AppearanceTypes, f?: ComputedFeature) => {
-    const [C, config] = components[k] ?? [];
-    if (!C || (f && !f[k]) || (config.noLayer && !f) || (config.noFeature && f)) {
-      return null;
+    // Check if the component output is already cached
+    const cachedComponent = CACHED_COMPONENTS.get(componentId);
```

Comment:
Question: If property is changed dynamically, is component re-rendered?

Reply:
It won't, I missed the property https://github.com/reearth/reearth-web/pull/608/commits/df1bdf392206201538d4d9cb4caaffde2361a790 should fix it.